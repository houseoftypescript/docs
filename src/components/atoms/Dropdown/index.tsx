import React from 'react';
import Button from '@mui/material/Button';
import Menu from '@mui/material/Menu';
import MenuItem from '@mui/material/MenuItem';

export type DropdownProps = {
  icon: React.ReactNode;
  title: string;
  items?: { children: React.ReactNode }[];
};

const Dropdown: React.FC<DropdownProps> = ({ icon, title, items = [] }) => {
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const open = Boolean(anchorEl);

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <div>
      <Button
        variant="contained"
        id="dropdown-button"
        aria-controls={open ? 'dropdown-menu' : undefined}
        aria-haspopup="true"
        aria-expanded={open ? 'true' : undefined}
        onClick={handleClick}
      >
        <div className="block md:hidden">{icon}</div>
        <div className="hidden md:block">{title}</div>
      </Button>
      {items.length > 0 && (
        <Menu
          id="dropdown-menu"
          anchorEl={anchorEl}
          open={open}
          onClose={handleClose}
          MenuListProps={{
            'aria-labelledby': 'dropdown-button',
          }}
        >
          {items.map((item, index) => {
            return (
              <MenuItem key={`menu-item-${index}`} onClick={handleClose}>
                {item.children}
              </MenuItem>
            );
          })}
        </Menu>
      )}
    </div>
  );
};

export default Dropdown;
