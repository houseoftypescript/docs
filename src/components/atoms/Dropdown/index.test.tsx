import Dropdown from '.';
import { render } from '@testing-library/react';

describe('Dropdown', () => {
  test('render default', () => {
    const wrapper = render(<Dropdown icon={<></>} title="title" />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
