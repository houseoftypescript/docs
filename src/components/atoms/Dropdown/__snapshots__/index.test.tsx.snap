// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown render default 1`] = `
<div>
  <div>
    <button
      aria-haspopup="true"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
      id="dropdown-button"
      tabindex="0"
      type="button"
    >
      <div
        class="block md:hidden"
      />
      <div
        class="hidden md:block"
      >
        title
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
